include/mbgl/style/expression/match.hpp

```diff
@@ -15,15 +15,14 @@ class Match : public Expression {
 public:
     using Branches = std::unordered_map<T, std::shared_ptr<Expression>>;
 
-    Match(type::Type type_,
+    Match(const type::Type& type_,
           std::unique_ptr<Expression> input_,
           Branches branches_,
-          std::unique_ptr<Expression> otherwise_
-    ) : Expression(Kind::Match, type_),
```

Comment:
move type, keep value semantics?

Reply:
It complains: `the parameter 'type_' is copied for each invocation but only used as a const reference`. I guess it is because type::Type doesn't have a move ctor.